src/ryu.jl

```diff
@@ -103,11 +103,12 @@ function get_engineering_string(x::AbstractFloat, precision::Integer)
     end
 
     buf = IOBuffer()
+    negative_base_compensation = base_digits[1] == '-' ? 1 : 0
```

Comment:
I could not find a better solution to compensate for an extra index in case of negative numbers,